src/context/WalletProvider/NativeWallet/components/LegacyLogin.tsx

```diff
@@ -41,36 +36,68 @@ export const LegacyLogin = () => {
     handleSubmit,
     register,
     formState: { errors, isSubmitting },
+    reset,
   } = useForm({ shouldUnregister: true })
 
   const translate = useTranslate()
 
-  const isLoginError = (err: any): err is LoginError => err && typeof err.message === 'string'
+  const isLoginError = (err: any): err is LoginError =>
+    typeof err?.response?.data?.error?.msg === 'string' && typeof err.response.status === 'number'
 
-  const onSubmit = async (values: FieldValues) => {
-    try {
-      // @TODO: Replace this with API call when backend is ready
-      const { mnemonic } = await Promise.resolve({ mnemonic: DUMMY_MNEMONIC })
+  const isDecryptionError = (err: any): err is Error =>
+    typeof err?.message === 'string' && err.message.startsWith('Native wallet decryption failed')
 
-      if (!bip39.validateMnemonic(mnemonic)) {
-        // @TODO: Should we send an error to the user?
-      }
+  type LoginResponse = {
+    success: boolean
+    data: string
+    error: any
+  }
 
-      // TODO: use response to create wallet vault.
+  const MIGRATE_URL = getConfig().REACT_APP_WALLET_MIGRATION_URL
+
+  const onSubmit = async (values: FieldValues) => {
+    try {
+      const hashedPassword = await getPasswordHash(values.email, values.password)
+      const { data: response } = await axios.post<LoginResponse>(`${MIGRATE_URL}`, {
+        email: values.email,
+        password: hashedPassword,
+        twoFactorCode: values.twoFactorCode || undefined,
+      })
+      const { data } = response
+      const encryptedWallet = data
```

Comment:
```suggestion
      const { data: encryptedWallet } = response
```
I would have done this instead of a new line but maybe there is a reason behind

Reply:
Agreed. Done.